app/src/components/adminConsole/currentCup/AdminCurrentSnapsDisplay.tsx

```diff
@@ -1,11 +1,28 @@
-import React from "react";
-import { MessageDisplay } from "../AdminConsoleStyles";
+import React, { useEffect, useState } from "react";
+import { getCurrentCupName } from "../../../firebase/cups/CupService";
+import { MessageDisplay, CupNameDisplay } from "../AdminConsoleStyles";
+import CurrentCupOptionsBar from "./CurrentCupOptionsBar";
+
 const AdminCurrentSnapsDisplay: React.FunctionComponent = (props: {
     isCup: Boolean;
+    isOpen: Boolean;
 }) => {
+    const [currentCupName, setCurrentCupName] = useState<String>("");
+    useEffect(() => {
+        getCurrentCupName()
+            .then((res: String) => {
+                setCurrentCupName(res);
+            })
+            .catch((e) => console.log(e));
+    }, [getCurrentCupName, setCurrentCupName]);
```

Comment:
It's very odd that we're going to the database for this, when we're passing the other properties of the cup in as props.

Reply:
I've merged the PR so it doesn't get impacted by any changes I make over the weekend. However, please look at incorporating this into further work you do.